file.go

```diff
@@ -75,14 +78,35 @@ func (ft *fileOutputTransport) Stop() {
 	ft.wg.Wait()
 }
 
-func (ft *fileOutputTransport) flushAll() {
+func (ft *fileOutputTransport) flushAll() error {
 	if ft.terminated {
-		return
+		return nil
+	}
+
+	fileList, _ := os.ReadDir(ft.path)
+	sort.Slice(fileList,
+		func(x int, y int) bool {
+			f1, err1 := os.Stat(fileList[x].Name())
+			f2, err2 := os.Stat(fileList[y].Name())
+			if err1 != nil || err2 != nil {
```

Comment:
Sort order would be unstable in this case. It's an edge case, but I think what you want to do is this:

```
if err1 != nil && err2 != nil {
    return false;
} else if err1 != nil {
    return false;
} else if err2 != nil {
    return true;
}
```